extension/app/src/lib/dust_api.ts

```diff
@@ -0,0 +1,22 @@
+import { DustAPI } from "@dust-tt/client";
+import { useAuth } from "@extension/components/auth/AuthProvider";
+
+export const useDustAPI = () => {
+  const { token, isAuthenticated, isUserSetup, user, workspace } = useAuth();
+
+  if (!isAuthenticated || !isUserSetup || !user || !workspace || !token) {
+    throw new Error("Not authenticated");
+  }
+
+  return new DustAPI(
+    {
+      url: process.env.DUST_DOMAIN ?? "https://dust.tt",
```

Comment:
NIT: I'm always uncomfortable setting prod env per default. Maybe we can throw if we don't have both vars? We do if not authenticated. WDYT?